### PR TITLE
Pmi

### LINKED ISSUE #44: Crud PMI
Excluding future admin authorization, make the following CRUD endpoints for /education:
- [ ] GET all
- [ ] GET all Happy Path
- [ ] GET by id
- [ ] GET by id Happy Path

- [ ] POST
- [ ] POST Happy Path Test belongs  
- [ ] PUT
- [ ] PUT Happy Path Test
- [ ] DELETE
- [ ] DELETE Happy Path Test

### LINKED ISSUE #47: Model PMI
- [ ] Add field for state
- [ ]  Add field for tax rate
- [ ]  Add field for average property tax

### PR DESCRIPTION
**Pull request by Arique1104**
# This PR:
---Adds model and CRUD functionality for PMI

## Changes
- _PMI stands for Principal Mortgage Insurance_
- _Based on your credit score and the amount of money you are putting down (or downpayment percentage)_
- [x] Adds code for functionality
- [x] Adds a test
- [ ] Updates Documentation/ReadMe
- [ ] Adds dependencies
- [ ] Refactors code
- [ ] Fixes a bug

## Screenshots
- _If applicable, use animated gif)_

## Checklist
**Please check off what you can for this PR:**
- [x] This code passes all tests on local machine
- [x] This code is DRY
- [x] This code follows OOP
- [x] This code, if updates to functionality were made, updates all corresponding files

**The "I have" section is a double check for PR specific information:**
- [ ] I have tagged reviewers for this PR
- [ ] I have checked what branch I am merging into (current expectation is development)
- [ ] I have added this PR to the project
- [ ] I have added this PR to the correct milestone
- [ ] I have updated documenation if changes were made that affect setup/running

## Linked tickest / issues 
Fixes #
Closes #47 
Closes #44

### CODE REVIEW

File: api/controller/PmiController.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
+from flask import Response, request
+from api.models.pmi import Pmi
+from mongoengine.errors import FieldDoesNotExist, NotUniqueError, DoesNotExist, ValidationError, InvalidQueryError
+import pry
+
+class PmiController():
+
```
Comment (priyapower) [inline api/controller/PmiController.py:7]: Do you mind putting a comment about which HTTP verbs each function represents

File: api/controller/PmiController.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            "data": {
+                "id": 'nil',
+                "confirmation": {
+                    "info": "To see this record's deletion response, please do a GET request using the url",
+                    "url": f'/api/v1/pmi{id}'
+                }
+            }
+        }, 200
+
+pmicontroller = PmiController()
```
Comment (priyapower) [inline api/controller/PmiController.py:94]: This controller looks great!

File: api/controller/PmiController.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        id = pmi.id
+        return {
+            "data": {
+                "id": str(id),
+                "confirmation": {
+                    "info": 'To see this new record, please do a GET request using the url',
+                    "url": f'/api/v1/pmi/{id}'
+                }
+            }
+        }, 200
```
Comment (priyapower) [inline api/controller/PmiController.py:21]: POST status code will probably update to 201

File: api/controller/PmiController.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        pmi.update(**body)
+        return {
+            "data": {
+                "id": str(id),
+                "confirmation": {
+                    "info": "To see this record's updated response, please do a GET request using the url",
+                    'url': f'/api/v1/pmi/{id}'
+                }
+            }
+        }, 200
```
Comment (priyapower) [inline api/controller/PmiController.py:79]: PUT status code will probably be 202

File: api/controller/PmiController.py [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        pmi.delete()
+        return {
+            "data": {
+                "id": 'nil',
+                "confirmation": {
+                    "info": "To see this record's deletion response, please do a GET request using the url",
+                    "url": f'/api/v1/pmi{id}'
+                }
+            }
+        }, 200
```
Comment (priyapower) [inline api/controller/PmiController.py:92]: DELETE status code will probably be 204

File: api/models/pmi.py [hunk 6]
```diff
@@ -0,0 +1,12 @@
+from database.db import db
```
Comment (priyapower) [inline api/models/pmi.py:1]: This model looks great!

File: api/routes/api_routes.py [hunk 7]
```diff
@@ -39,3 +40,23 @@ def update_report(id):
 @api.route('/api/v1/report/<id>', methods=['DELETE'])
 def destroy_report(id):
     return reportcontroller.destroy_report(id)
+
+@api.route('/api/v1/pmi/<id>', methods=['GET'])
```
Comment (priyapower) [inline api/routes/api_routes.py:44]: Great job with these routes. 
Question? Do we want to nest any of the models that are necessary for report under report. Like `/api/v1/report/pmi`?

File: database/pmi.py [hunk 8]
```diff
@@ -0,0 +1,44 @@
+downpayment_zero = {
```
Comment (priyapower) [inline database/pmi.py:1]: Great job with abstracting large datasets into their own file!

File: pmi.md [hunk 9]
```diff
@@ -0,0 +1,161 @@
+
```
Comment (priyapower) [inline pmi.md:1]: I like your notes documentation here

File: tests/test_pmi_crud.py [hunk 10]
```diff
@@ -0,0 +1,149 @@
+import pry
```
Comment (priyapower) [inline tests/test_pmi_crud.py:1]: Thanks for the happy path testing! I look forward to seeing the sad path testing

**Review by priyapower — APPROVED** [review]
Looks great

**Review by EHale64 — APPROVED** [review]

**Pull request merged**